### PR TITLE
This adds a simple input interface

### PR DESCRIPTION
**Pull request by Izaakwltn**
Formulador-lite is a simple input interface, for converting inputs such as 1+2 or [1/2]+[2/3] into formulador drawings.

Currently fixing the block system to accommodate larger formulas, and fixing the exponent system to allow for closed-parentheses.

### CODE REVIEW

File: formulador-lite/examples.lisp [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+;;;;------------------------------------------------------------------------
+
+FORMULADOR-LITE> (simple-draw "1/[[1/h] + [1/a] + [1/r] + [1/m] + [1/o] + [1/n] + [1/i] + [1/c]]")
+#<FORMULADOR:CANVAS 
++---------------------------------------------------------------+
+|                               1                               |
+|---------------------------------------------------------------|
+|   1       1       1       1       1       1       1       1   |
+| ----- + ----- + ----- + ----- + ----- + ----- + ----- + ----- |
+|  |h|     |a|     |r|     |m|     |o|     |n|     |i|     |c|  |
```
Comment (notmgsk) [inline formulador-lite/examples.lisp]: Why are these wrapped with bars?

**Comment by notmgsk** [comment]
Nice idea :)

File: formulador-lite/examples.lisp (hunk 1 above)
Comment (Izaakwltn) [reply formulador-lite/examples.lisp]: That luckily was a relic of previous tests- I double checked today and it worked fine:

+-----------------------------------------------+
|                       1                       |
|-----------------------------------------------|
|  1     1     1     1     1     1     1     1  |
| --- + --- + --- + --- + --- + --- + --- + --- |
|  h     a     r     m     o     n     i     c  |
+-----------------------------------------------+

**Comment by Izaakwltn** [comment]
Thank you! Any suggestions on what to add to it?

File: formulador-lite.asd [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+(asdf:defsystem #:formulador-lite
+  :author "Izaak Walton <izaakw@protonmail.com>"
+  :license "BSD 3-clause (see LICENSE)"
+  :description "Simplified formula interface."
+  :long-description "A simple input interface for Formulador."
+  :depends-on (#:formulador #:alexa)
+  :serial t
+  :components ((:module "formulador-lite"
+		:serial t
```
Comment (stylewarning) [inline formulador-lite.asd]: indentation wonky due to tabs?

File: formulador-lite.asd [hunk 3]
```diff
@@ -0,0 +1,26 @@
+;;;;formulador-lite.asd
+;;;;
+;;;;Copyright (c) 2021 Izaak Walton
+
+(defpackage #:formulador-lite-asd
```
Comment (stylewarning) [inline formulador-lite.asd]: no need for this defpackage stuff

File: formulador-lite.asd [hunk 4]
```diff
@@ -0,0 +1,26 @@
+;;;;formulador-lite.asd
```
Comment (stylewarning) [inline formulador-lite.asd]: spaces after the semicolon group

```
;;;; formulator-lite.asd
````

File: formulador-lite.asd~ [hunk 5]
```diff
@@ -0,0 +1,26 @@
+;;;;formulador-lite.asd
```
Comment (stylewarning) [inline formulador-lite.asd~]: file accidentally committed

File: formulador-lite/blocks.lisp [hunk 6]
```diff
@@ -0,0 +1,89 @@
+;;;;formulador-lite/blocks.lisp
+;;;;
+;;;; Copyright (c) 2021 Izaak Walton
+
+(in-package :formulador-lite)
```
Comment (stylewarning) [inline formulador-lite/blocks.lisp]: I'm partial to `#:formulador-lite`

File: formulador-lite/blocks.lisp [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+;;;;------------------------------------------------------------------------
+;;;;Generating a list of blocks and operators
```
Comment (stylewarning) [inline formulador-lite/blocks.lisp]: use triple semicolon `;;;` and no `-------`

File: formulador-lite/tests.lisp [hunk 8]
```diff
@@ -0,0 +1,58 @@
+;;;;tests.lisp
+
+(in-package :formulador-lite)
```
Comment (stylewarning) [inline formulador-lite/tests.lisp]: tests ought to go into a `fiasco` package

File: formulador-lite/unit-detection.lisp [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+;;;;------------------------------------------------------------------------
+;;;;Formula unit detection:
+;;;;------------------------------------------------------------------------
+
+;;;;------------------------------------------------------------------------
+;;;;Bracket and Block Detection
+;;;;------------------------------------------------------------------------
+(defun detect-brack (lexed-line)
+  "Detects the beginning of a bracketed block."
+  (equal (car (first lexed-line)) ':left-brack))
```
Comment (stylewarning) [inline formulador-lite/unit-detection.lisp]: `eq`

File: formulador-lite/unit-detection.lisp [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+
+(defun brack-length (lexed-line counter)
+  "Determines the number of elements before the end of a bracketed block."
+  (cond ((detect-end-brack lexed-line) counter)
```
Comment (stylewarning) [inline formulador-lite/unit-detection.lisp]: cond branches on their own lines

File: formulador-lite/unit-detection.lisp [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+	(t (brack-length (rest lexed-line) (+ counter 1)))))
+
+(defun detect-end-brack (lexed-line)
+  "Detects the end of a bracketed block."
+  (equal (car (first lexed-line)) ':right-brack))
```
Comment (stylewarning) [inline formulador-lite/unit-detection.lisp]: `eq` here as well, etc.

File: formulador-lite/parentheses.lisp [hunk 12]
```diff
@@ -0,0 +1,41 @@
+;;;;formulador-lite/parentheses.lisp
+;;;;
+;;;;Copyright (c) 2021 Izaak Walton
+
+(in-package :formulador-lite)
+
+
+(defun deep-length (list)
```
Comment (stylewarning) [inline formulador-lite/parentheses.lisp]: maybe `count-atoms` or `tree-size`?

File: formulador-lite/drawing.lisp [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+(in-package :formulador-lite)
+
+
+;;;;;------------------------------------------------------------------------
+;;;;;The simple drawing command: (simple-draw "1+2")
+;;;;;------------------------------------------------------------------------
+(defun simple-draw (formula-string)
+  "Given a formula-string, returns pretty-printed formulador formula."
+  (formulador::draw
+   (eval (first (block-cycle (block-list (lex-line formula-string)))))))
```
Comment (stylewarning) [inline formulador-lite/drawing.lisp]: you can avoid `eval` by building the structures in the first place, instead of making quoted lists.

**Review by stylewarning — CHANGES_REQUESTED** [review]
some suggestions to begin with, without getting to deep into implementation details

File: formulador-lite.asd (hunk 2 above)
Comment (Izaakwltn) [reply formulador-lite.asd]: I've fixed the formatting, just figuring out evaluation issues- is it common practice to "Resolve conversation" as I fix stuff, or to leave it until I push an updated request?

**Comment by Izaakwltn** [comment]
I've updated the code to build drawings organically instead of using quoted lists, and have updated the block-unit system. Fiasco is giving me an odd bug- (run-package-tests) returns "invalid number of arguments: 9." I don't necessarily need tests for this project but any illumination would be appreciated.

**Pull request closed without merging**